rasa/core/policies/ted_policy.py

```diff
@@ -0,0 +1,642 @@
+        # the scale of how important is to minimize the maximum similarity
+        # between embeddings of different labels
+        NEGATIVE_MARGIN_SCALE: 0.8,
+        # dropout rate for dial nn
+        DROP_RATE_DIALOGUE: 0.1,
+        # dropout rate for bot nn
+        DROP_RATE_LABEL: 0.0,
+        # dropout rate for attention
+        DROP_RATE_ATTENTION: 0,
+        # visualization of accuracy
+        # how often calculate validation accuracy
+        EVAL_NUM_EPOCHS: 20,  # small values may hurt performance
+        # how many examples to use for hold out validation set
+        EVAL_NUM_EXAMPLES: 0,  # large values may hurt performance
+    }
+    # end default properties (DOC MARKER - don't remove)
+
+    @staticmethod
+    def _standard_featurizer(max_history: Optional[int] = None) -> TrackerFeaturizer:
+        if max_history is None:
+            return FullDialogueTrackerFeaturizer(LabelTokenizerSingleStateFeaturizer())
+        else:
+            return MaxHistoryTrackerFeaturizer(
+                LabelTokenizerSingleStateFeaturizer(), max_history=max_history
+            )
+
+    def __init__(
+        self,
+        featurizer: Optional[TrackerFeaturizer] = None,
+        priority: int = DEFAULT_POLICY_PRIORITY,
+        max_history: Optional[int] = None,
+        model: Optional[RasaModel] = None,
+        **kwargs: Dict[Text, Any],
+    ) -> None:
+        """Declare instance variables with default values"""
+
+        if not featurizer:
+            featurizer = self._standard_featurizer(max_history)
+
+        super().__init__(featurizer, priority)
+
+        self._load_params(**kwargs)
+
+        self.model = model
+
+        self._label_data = None
+        self.data_example = None
+
+    def _load_params(self, **kwargs: Dict[Text, Any]) -> None:
+        self.config = copy.deepcopy(self.defaults)
+        self.config.update(kwargs)
+
+        self.config = train_utils.check_deprecated_options(self.config)
+
+        self.config = train_utils.update_similarity_type(self.config)
+
+        if self.config[EVAL_NUM_EPOCHS] == -1:
+            # magic value -1 is used to set evaluation to number of epochs
+            self.config[EVAL_NUM_EPOCHS] = self.config[EPOCHS]
+        elif self.config[EVAL_NUM_EPOCHS] < 1:
+            raise ValueError(
+                f"'{EVAL_NUM_EXAMPLES}' is set to '{self.config[EVAL_NUM_EPOCHS]}'. "
+                f"Only values > 1 are allowed for this configuration value."
+            )
+
+    # data helpers
+    # noinspection PyPep8Naming
+    @staticmethod
+    def _label_ids_for_Y(data_Y: np.ndarray) -> np.ndarray:
+        """Prepare Y data for training: extract label_ids."""
+
+        return data_Y.argmax(axis=-1)
+
+    # noinspection PyPep8Naming
+    def _label_features_for_Y(self, label_ids: np.ndarray) -> np.ndarray:
+        """Prepare Y data for training: features for label_ids."""
+
+        # full dialogue featurizer is used
+        if len(label_ids.shape) == 2:
```

Comment:
if you make `len(label_ids.shape) == 2` a function or a named variable you can get rid of the comment
```suggestion
        is_full_dialogue_featurizer_used = len(label_ids.shape) == 2
        if is_full_dialogue_featurizer_used == 2:
```